engine/src/flutter/lib/web_ui/lib/src/engine/semantics/semantics.dart

```diff
@@ -756,6 +764,31 @@ abstract class SemanticRole {
     removeAttribute('aria-controls');
   }
 
+  /// Applies the current [SemanticsObject.validationResult] to the DOM managed
+  /// by this role.
+  ///
+  /// The default implementation applies the `aria-invalid` attribute to the
+  /// root [SemanticsObject.element]. Specific role implementations may prefer
+  /// to apply it to different elements, depending on their use-case. For
+  /// example, a text field may want to apply it on the underlying `<input>`
+  /// element.
+  void updateValidationResult() {
+    updateAriaInvalid(semanticsObject.element, semanticsObject.validationResult);
+  }
+
+  /// Converts [validationResult] to its ARIA value and sets it as the `aria-invalid`
+  /// attribute of the given [element].
+  ///
+  /// If [validationResult] is null, removes the `aria-invalid` attribute from
+  /// the element.
+  static void updateAriaInvalid(DomElement element, ui.SemanticsValidationResult validationResult) {
+    if (validationResult != ui.SemanticsValidationResult.none) {
+      element.setAttribute('aria-invalid', validationResult.toAriaValue());
```

Comment:
isn't the aria-invalid a boolean value?
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-invalid

Reply:
See the "Values" section: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-invalid#values

Reply:
ah, I should scroll it down further. thanks